src/app/components/layouts/tag/terra-tag.component.spec.ts

```diff
@@ -1,11 +1,255 @@
+
+        expect(Object.entries(tagDiv.classes).length).toBeGreaterThan(0);
+        expect(tagDiv.classes[customClass]).toBe(true);
+    });
+
+    it('should show tag icon depending on inputIsTaggable', () =>
+    {
+        let iconElement:DebugElement = tagDiv.query(By.css('span.tag-icon'));
+        expect(iconElement).toBeFalsy();
+
+        component.inputIsTaggable = true;
+
+        fixture.detectChanges();
+
+        iconElement = tagDiv.query(By.css('span.tag-icon'));
+
+        expect(iconElement).toBeTruthy();
+    });
+
+    it('should set color style to tag icon depending on inputColor', () =>
+    {
+        component.inputIsTaggable = true;
+        fixture.detectChanges();
+
+        let iconElement:DebugElement = tagDiv.query(By.css('span.tag-icon'));
+
+        expect(iconElement.styles['color']).toBeFalsy(); // style is not present
+
+        component.inputColor = tagTwo.color;
+
+        fixture.detectChanges();
+
+        iconElement = tagDiv.query(By.css('span.tag-icon'));
+
+        // getting access to protected/private methods
+        expect(iconElement.styles['color']).toEqual(component['getColor']()); // style is present and equals #ffffff or #000000
+    });
+
+    it('should set classes to tag icon depending on inputIsTagged', () =>
+    {
+        component.inputIsTaggable = true;
+        component.inputIsTagged = true;
+
+        fixture.detectChanges();
+
+        let iconElement:DebugElement = tagDiv.query(By.css('span.tag-icon'));
+
+        expect(iconElement.classes['isTagged']).toBe(true);
+        expect(iconElement.classes['icon-ticket_prio1']).toBe(true);
+        expect(iconElement.classes['icon-ticket_prio8']).toBe(false);
+
+        component.inputIsTagged = false;
+
+        fixture.detectChanges();
+
+        expect(iconElement.classes['isTagged']).toBe(false);
+        expect(iconElement.classes['icon-ticket_prio1']).toBe(false);
+        expect(iconElement.classes['icon-ticket_prio8']).toBe(true);
+    });
+
+    it('should set color style to tag text depending on inputColor', () =>
+    {
+        component.inputIsTaggable = true;
+        fixture.detectChanges();
+
+        let textElement:DebugElement = tagDiv.query(By.css('span.tag-text'));
+
+        expect(textElement.styles['color']).toBeFalsy(); // style is not present
+
+        component.inputColor = tagTwo.color;
+
+        fixture.detectChanges();
+
+        // getting access to protected/private methods
+        expect(textElement.styles['color']).toEqual(component['getColor']()); // style is present and equals #ffffff or #000000
+    });
+
+    it('should set text depending on inputBadge', () =>
+    {
+        component.inputBadge = name;
+
+        fixture.detectChanges();
+
+        let textElement:DebugElement = tagDiv.query(By.css('span.tag-text'));
+        let text:HTMLSpanElement = textElement.nativeElement;
+
+        expect(text.innerHTML).toEqual(name);
+
+        component.inputBadge = null;
```

Comment:
das machst du bereits in 'should set text depending on name'

Reply:
ja aber vor jedem test wird die component neu erstellt
eigentlich brauch ich es nicht setzen